src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -40,10 +46,23 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 			return nil
 		}); err != nil {
 			log.Error(err, "Failed database transaction to check cluster member role")
+			txnRetries++
 		}
+
+		if txnRetries >= a.config.MaxPreRemoveHookDBTxnRetries {
+			return false, ErrMaxRemoveTxnRetries
+		}
```

Comment:
I would hardcode this to whatever you would have made the default value if it were configurable.

Reply:
I think it might. imo it's harmless to keep this around as a config. 
for others seeing this: more discussions about this one on MM